### ISSUE TITLE
Java 10 compatibility: cannot decode verbose time (javax/xml/bind/DatatypeConverter)

### ISSUE THREAD

**Issue by timothypratley** [cognitect/transit-java#28]
When using Java 10 (or 9), transit cannot decode verbose time. such as `~t2018-12-31T00:00:00.000Z`

Symptom:
```
  actual: java.lang.RuntimeException: java.lang.NoClassDefFoundError: javax/xml/bind/DatatypeConverter
 at com.cognitect.transit.impl.ReaderFactory$ReaderImpl.read (ReaderFactory.java:114)
    cognitect.transit$read.invokeStatic (transit.clj:319)
```

Minimal repro example:
```
(ns ttt.core
  (:require [cognitect.transit :as transit]
            [clojure.java.io :as io]))

(defn transit-decode [body]
  (let [in (io/input-stream (.getBytes body))
        reader (transit/reader in :json)]
    (transit/read reader)))

(defn foo []
  (transit-decode "{\"hi\":\"~t2018-12-31T00:00:00.000Z\"}"))
```

Cause:

com.cognitect.transit.impl/ReaderHandlers.java, line 248 uses `javax.xml.bind.DatatypeConverter.parseDateTime`:
```
    public static class VerboseTimeReadHandler implements ReadHandler<Object, String> {

        @Override
        public Object fromRep(String rep) {
            Calendar t = javax.xml.bind.DatatypeConverter.parseDateTime(rep);
            t.setTimeZone(TimeZone.getTimeZone("Zulu"));
            return t.getTime();
        }
    }
```
Java 10 (and 9) introduces modules, and javax.xml.bind is no longer included by default.
https://www.deps.co/blog/how-to-upgrade-clojure-projects-to-use-java-9/

Suggestion:

Use a different method to read the time which is compatible with Java 8 and 10?
(to avoid having to use Java flags to enable the javax.xml.bind module).

**Comment by puredanger** [comment]
I'll take a look at this when I get a chance.

**Comment by timothypratley** [comment]
Hi @puredanger,
I think you are correct that `clojure.instant/parse-timestamp` can be used as a replacement; shall I make a pull-request to that effect?

**Comment by puredanger** [comment]
Sorry, we don't take PRs on this project, but I will get to it.

**Comment by timothypratley** [comment]
Okidoki thank you :)

**Comment by timothypratley** [comment]
Hi, I am shamelessly checking in to see if you have time to address this... no big deal if not.

**Comment by puredanger** [comment]
Added explicit dep to resolve this in transit-java 0.8.337 (also available via transit-clj 0.8.313).

**Comment by timothypratley** [comment]
Thank you Alex!!! :)
:+1:
